spyder/preferences/tests/test_shorcuts.py

```diff
@@ -9,29 +9,207 @@
 """
 
 import os
+import sys
 
 # Test library imports
 import pytest
+from qtpy.QtCore import Qt
 
 # Local imports
-from spyder.preferences.shortcuts import ShortcutsTable
+from spyder.config.main import CONF
+from spyder.preferences.shortcuts import (
+    ShortcutsTable, ShortcutEditor, NO_WARNING, SEQUENCE_CONFLICT,
+    INVALID_KEY, SEQUENCE_EMPTY)
 
+
+# ---- Qt Test Fixtures
 @pytest.fixture
-def setup_shorcuts(qtbot):
+def shorcut_table_bot(qtbot):
     """Set up shortcuts."""
-    widget = ShortcutsTable()
-    qtbot.addWidget(widget)
-    return widget
+    shorcut_table = ShortcutsTable()
+    qtbot.addWidget(shorcut_table)
+    return shorcut_table, qtbot
+
+
+@pytest.fixture
+def shortcut_editor_bot(shorcut_table_bot):
+    shorcut_table, qtbot = shorcut_table_bot
+    shortcuts = shorcut_table.source_model.shortcuts
+
+    def _create_bot(context, name):
+        sequence = CONF.get('shortcuts', "{}/{}".format(context, name))
+        shortcut_editor = ShortcutEditor(
+            shorcut_table, context, name, sequence, shortcuts)
+        qtbot.addWidget(shortcut_editor)
+        shortcut_editor.show()
+        return shortcut_editor, qtbot
```

Comment:
I don't like this style of fixtures: we should make our fixtures to return the single object we want to test and not the tuple of `widget, qtbot`. Instead `qtbot` needs to be passed as an additional fixture to every test, as I point out in a test below.

So please change this.